go/parquet/pqarrow/schema.go

```diff
@@ -948,7 +957,21 @@ func getNestedFactory(origin, inferred arrow.DataType) func(fieldList []arrow.Fi
 func applyOriginalStorageMetadata(origin arrow.Field, inferred *SchemaField) (modified bool, err error) {
 	nchildren := len(inferred.Children)
 	switch origin.Type.ID() {
-	case arrow.EXTENSION, arrow.SPARSE_UNION, arrow.DENSE_UNION:
+	case arrow.EXTENSION:
+		extType := origin.Type.(arrow.ExtensionType)
+		modified, err = applyOriginalStorageMetadata(arrow.Field{
+										 Type: extType.StorageType(), 
+										 Metadata: arrow.NewMetadata(
+																[]string{ipc.ExtensionTypeKeyName, ipc.ExtensionMetadataKeyName},
+																[]string{extType.ExtensionName(), extType.Serialize()}),
+							 }, inferred)
+		if err != nil {
+				return
+		}
```

Comment:
can you run `gofmt` please to fix the spacing here?

Reply:
Fixed